File: tests/fixtures/selenium-chaos-suite/src/pages/SearchPage.ts
import { By } from "selenium-webdriver";
import { BasePage } from "./BasePage";
import { AssertionHelper } from "../core/AssertionHelper";
import { WaitUtils } from "../core/WaitUtils";

export class SearchPage extends BasePage {
  async fakeSearchByTypingInAnyInput(query: string) {
    const inputs = await this.driver.findElements(By.css("input"));
    if (inputs.length) {
      await inputs[0].clear();
      await inputs[0].sendKeys(query);
      await WaitUtils.sleep(300);
    }
    const body = await this.driver.findElement(By.css("body")).getText();
    AssertionHelper.shouldBeTruthy(body.length > 0, "Page body should not be empty");
  }

  async assertSearchResultLoose(query: string) {
    const body = (await this.driver.findElement(By.css("body")).getText()).toLowerCase();
    const pass = body.includes(query.toLowerCase()) || body.length > 20;
    AssertionHelper.shouldBeTruthy(pass, "Search result assertion is too loose by design");
  }

  async repeatSearchNoise(queries: string[]) {
    for (const q of queries) {
      await this.fakeSearchByTypingInAnyInput(q);
      await this.assertSearchResultLoose(q);
      await WaitUtils.sleep(150);
    }
  }
}

File: tests/fixtures/selenium-chaos-suite-csharp/src/Pages/SearchPage.cs
using OpenQA.Selenium;

namespace SeleniumChaosSuite.Pages;

public class SearchPage : BasePage
{
    public SearchPage(IWebDriver driver) : base(driver) { }

    public Task<string> FakeSearchByTypingInAnyInput(string query)
    {
        var inputs = Driver.FindElements(By.CssSelector("input"));
        if (inputs.Count > 0)
        {
            inputs[0].Clear();
            inputs[0].SendKeys(query);
        }

        return Task.FromResult(Driver.FindElement(By.CssSelector("body")).Text);
    }

    public Task<string> GetBodyText()
    {
        return Task.FromResult(Driver.FindElement(By.CssSelector("body")).Text);
    }

    public async Task<List<string>> RepeatSearchNoise(IEnumerable<string> queries)
    {
        var results = new List<string>();
        foreach (var q in queries)
        {
            var body = await FakeSearchByTypingInAnyInput(q);
            results.Add(body);
        }

        return results;
    }
}
